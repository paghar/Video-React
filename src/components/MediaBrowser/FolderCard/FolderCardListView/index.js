//@flow
import React from 'react';
import { Display2, Display5 } from 'theme/fonts';
import { Generals } from 'theme/colors';
import { getListViewDate } from 'helpers/time';
import {
  ItemWrapper,
  NameWrapper,
  FileNumberLabel,
  FolderNameLabel,
  FileDialogs,
  StyledIcon,
  MenuToggler,
  FileUploader,
  FileSize,
  FileDuration,
  FileDate,
  FileStatus,
  Duration,
} from './index.style';
type Props = {
  selected?: boolean,
  fileCount: number,
  name: string,
  uploadDate: number,
  uploader: string,
  size?: string,
  className?: string,
  dragging?: boolean,
};
const FolderCardListView = ({
  uploadDate,
  fileCount,
  name,
  className,
  uploader,
  size,
  selected,
  dragging,
}: Props) => {
  return (
    <ItemWrapper dragging={dragging}>
      <NameWrapper>
        <FolderNameLabel
          selected={selected}
          color={Generals.vii}
          font={Display2}
        >
          {name}
        </FolderNameLabel>
        <FileNumberLabel
          selected={selected}
          hoverable
          font={Display2}
          color={Generals.vi}
        >
          {fileCount} files
        </FileNumberLabel>
      </NameWrapper>
      <FileUploader selected={selected} color={Generals.vi} font={Display5}>
        {uploader}
      </FileUploader>
      <FileDialogs>
        <StyledIcon
          selected={selected}
          fontSize="12px"
          color={Generals.vi}
          hoverColor={Generals.vii}
          iconNumber={62161}
        />
      </FileDialogs>
      <FileSize selected={selected} color={Generals.vi} font={Display5}>
        {size}
      </FileSize>
      <FileDuration>
        <StyledIcon
          selected={selected}
          fontSize="12px"
          color={Generals.vi}
          hoverColor={Generals.vii}
          iconNumber={62161}
        />
      </FileDuration>

      <FileDate selected={selected} color={Generals.vi} font={Display5}>
        {getListViewDate(uploadDate)}
      </FileDate>
      <FileStatus>
        <StyledIcon
          selected={selected}
          fontSize="12px"
          color={Generals.vi}
          hoverColor={Generals.vii}
          iconNumber={62161}
        />
      </FileStatus>
      <MenuToggler
        backgroundColor="transparent"
        hoverBackgroundColor="transparent"
        hoverColor={Generals.vii}
        radius="0px"
        iconSize="12px"
        icon="61762"
        spacing="0"
        width="auto"
        height="auto"
        iconColor={Generals.vii}
        hoverIconColor={Generals.vii}
        selected={selected}
      />
    </ItemWrapper>
  );
};

export default FolderCardListView;
