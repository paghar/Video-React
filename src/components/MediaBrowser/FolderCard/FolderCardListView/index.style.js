//@flow
import styled from 'styled-components';
import { Generals } from 'theme/colors';
import Typography from 'components/shared/Typography';
import Button from 'components/shared/Button';
import Icon from 'components/shared/Icon';

export const NameWrapper = styled.div`
  margin-left: 9px;
  display: grid;
  align-self: center;
  flex-direction: column;

  @media (min-width: 1141px) {
    width: 213px;
  }
  @media (max-width: 1140px) {
    min-width: 78px;
    width: 20%;
  }
`;
export const FolderNameLabel = styled(Typography)`
  display: block;
  white-space: nowrap;
  pointer-events: none;
  overflow: hidden;
  text-overflow: ellipsis;
  color: ${({ selected }) => (selected ? Generals.viii : Generals.vii)};
`;
export const FileNumberLabel = styled(Typography)`
  outline: none;
  align-self: auto;
  pointer-events: none;
  margin: 0;
  color: ${({ selected }) => (selected ? Generals.vii : Generals.vi)};
`;

export const FileUploader = styled(Typography)`
  width: 113px;
  display: block;
  pointer-events: none;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  align-self: center;
  color: ${({ selected }) => (selected ? Generals.vii : Generals.vi)};
  @media (max-width: 1140px) {
    width: 12%;
    min-width: 68px;
  }
`;
export const FileDialogs = styled.div`
  display: flex;
  align-self: center;
  min-width: 46px;
`;
export const StyledIcon = styled(Icon)`
  color: ${({ selected }) => (selected ? Generals.vii : Generals.vi)};
  margin-left: 3px;
  pointer-events: none;
  margin-bottom: 5px;
  user-select: none;
`;
export const FileSize = styled(Typography)`
  color: ${({ selected }) => (selected ? Generals.vii : Generals.vi)};
  align-self: center;
  pointer-events: none;
  min-width: 42px;
`;
export const FileDuration = styled.div`
  width: 55px;
  display: flex;
  align-self: center;
  @media (max-width: 1140px) {
    display: none;
  }
`;
export const Duration = styled(Typography)`
  color: ${({ selected }) => (selected ? Generals.vii : Generals.vi)};
  align-self: center;
  pointer-events: none;
`;
export const FileDate = styled(Typography)`
  color: ${({ selected }) => (selected ? Generals.vii : Generals.vi)};
  display: flex;
  pointer-events: none;
  align-self: center;
  min-width: 86px;
`;

export const FileStatus = styled.div`
  width: 98px;
  align-self: center;
  display: flex;
`;
export const MenuToggler = styled(Button)`
  display: flex;
  visibility: ${({ selected }) => (selected ? 'visible' : 'hidden')};
  align-self: center;
  position: absolute;
  right: 12px;
  :hover {
    cursor: pointer;
  }
`;
export const ItemWrapper = styled.div`
  display: flex;
  width:calc(100% - 89px);
  justify-content: space-between;
  flex: 1 1 auto;
  :hover {
    ${FileDate},
    ${StyledIcon},
    ${FileSize},
    ${FileUploader},
    ${FileNumberLabel} {
      color: ${Generals.vii};
    }
    ${FolderNameLabel} {
      color: ${Generals.viii};
    }
    ${MenuToggler} {
      visibility:visible !important;
    }
  }
`;
