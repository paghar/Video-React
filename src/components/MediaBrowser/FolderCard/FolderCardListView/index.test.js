//@flow
import React from 'react';
import FolderCardListView from '.';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('<FolderCardListView />', () => {
  it('checks  UI', () => {
    const wrapper = renderer
      .create(
        <FolderCardListView
          selected={false}
          name="Hiking Experience"
          fileCount={3}
          uploader="Emily Parkson"
          size="12 MB"
          uploadDate={1553277015000}
        />
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('checks UI selected', () => {
    const wrapper = renderer
      .create(
        <FolderCardListView
          selected={true}
          name="Hiking Experience"
          fileCount={3}
          uploader="Emily Parkson"
          size="12 MB"
          uploadDate={1553277015000}
        />
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
