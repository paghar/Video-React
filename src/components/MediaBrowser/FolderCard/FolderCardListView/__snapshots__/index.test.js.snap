// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FolderCardListView /> checks  UI 1`] = `
.c2 {
  pointer-events: auto;
}

.c4 {
  pointer-events: none;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c8 {
  font-family: 'FontAwesome' !important;
  font-size: 12px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c17 {
  font-size: 12px;
  display: block;
  padding-right: 0;
  padding-left: 0px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c14 {
  direction: ltr;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  background: none;
  background-color: transparent;
  width: auto;
  height: auto;
  border-radius: 0px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c14:hover {
  background-color: transparent;
}

.c1 {
  margin-left: 9px;
  display: grid;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: block;
  white-space: nowrap;
  pointer-events: none;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c5 {
  outline: none;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  pointer-events: none;
  margin: 0;
}

.c6 {
  width: 113px;
  display: block;
  pointer-events: none;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  min-width: 46px;
}

.c9 {
  margin-left: 3px;
  pointer-events: none;
  margin-bottom: 5px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c10 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  pointer-events: none;
  min-width: 42px;
}

.c11 {
  width: 55px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  pointer-events: none;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  min-width: 86px;
}

.c13 {
  width: 98px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  visibility: hidden;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  position: absolute;
  right: 12px;
}

.c16:hover {
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: calc(100% - 89px);
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c0:hover .c15 {
  visibility: visible !important;
}

@media (min-width:1141px) {
  .c1 {
    width: 213px;
  }
}

@media (max-width:1140px) {
  .c1 {
    min-width: 78px;
    width: 20%;
  }
}

@media (max-width:1140px) {
  .c6 {
    width: 12%;
    min-width: 68px;
  }
}

@media (max-width:1140px) {
  .c11 {
    display: none;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
    >
      Hiking Experience
    </span>
    <span
      className="c4 c5"
    >
      3
       files
    </span>
  </div>
  <span
    className="c2 c6"
  >
    Emily Parkson
  </span>
  <div
    className="c7"
  >
    <span
      className="c8 c9"
      fontSize="12px"
    >
      
    </span>
  </div>
  <span
    className="c2 c10"
  >
    12 MB
  </span>
  <div
    className="c11"
  >
    <span
      className="c8 c9"
      fontSize="12px"
    >
      
    </span>
  </div>
  <span
    className="c2 c12"
  >
    22th-MAR-22:20
  </span>
  <div
    className="c13"
  >
    <span
      className="c8 c9"
      fontSize="12px"
    >
      
    </span>
  </div>
  <button
    className="c14 c15 c16"
    height="auto"
    radius="0px"
    width="auto"
  >
    <span
      className="c8 c17"
      fontSize="12px"
    >
      
    </span>
  </button>
</div>
`;

exports[`<FolderCardListView /> checks UI selected 1`] = `
.c2 {
  pointer-events: auto;
}

.c4 {
  pointer-events: none;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c8 {
  font-family: 'FontAwesome' !important;
  font-size: 12px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c17 {
  font-size: 12px;
  display: block;
  padding-right: 0;
  padding-left: 0px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c14 {
  direction: ltr;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  background: none;
  background-color: transparent;
  width: auto;
  height: auto;
  border-radius: 0px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c14:hover {
  background-color: transparent;
}

.c1 {
  margin-left: 9px;
  display: grid;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: block;
  white-space: nowrap;
  pointer-events: none;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c5 {
  outline: none;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  pointer-events: none;
  margin: 0;
}

.c6 {
  width: 113px;
  display: block;
  pointer-events: none;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  min-width: 46px;
}

.c9 {
  margin-left: 3px;
  pointer-events: none;
  margin-bottom: 5px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c10 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  pointer-events: none;
  min-width: 42px;
}

.c11 {
  width: 55px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  pointer-events: none;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  min-width: 86px;
}

.c13 {
  width: 98px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  visibility: visible;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  position: absolute;
  right: 12px;
}

.c16:hover {
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: calc(100% - 89px);
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c0:hover .c15 {
  visibility: visible !important;
}

@media (min-width:1141px) {
  .c1 {
    width: 213px;
  }
}

@media (max-width:1140px) {
  .c1 {
    min-width: 78px;
    width: 20%;
  }
}

@media (max-width:1140px) {
  .c6 {
    width: 12%;
    min-width: 68px;
  }
}

@media (max-width:1140px) {
  .c11 {
    display: none;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2 c3"
    >
      Hiking Experience
    </span>
    <span
      className="c4 c5"
    >
      3
       files
    </span>
  </div>
  <span
    className="c2 c6"
  >
    Emily Parkson
  </span>
  <div
    className="c7"
  >
    <span
      className="c8 c9"
      fontSize="12px"
    >
      
    </span>
  </div>
  <span
    className="c2 c10"
  >
    12 MB
  </span>
  <div
    className="c11"
  >
    <span
      className="c8 c9"
      fontSize="12px"
    >
      
    </span>
  </div>
  <span
    className="c2 c12"
  >
    22th-MAR-22:20
  </span>
  <div
    className="c13"
  >
    <span
      className="c8 c9"
      fontSize="12px"
    >
      
    </span>
  </div>
  <button
    className="c14 c15 c16"
    height="auto"
    radius="0px"
    width="auto"
  >
    <span
      className="c8 c17"
      fontSize="12px"
    >
      
    </span>
  </button>
</div>
`;
